core/src/test/scala/pickling/run/runtime-instantiation-static-picklers.scala

```diff
@@ -0,0 +1,219 @@
+package scala.pickling.runtime
+
+import org.scalatest.FunSuite
+import scala.pickling._, scala.pickling.Defaults._, json._
+
+case class Container[T](value: T)
+
+object RuntimeHelper {
+  /* Instantiate a class for an already generated `Unpickler` */
+  def getInstance[T](className: String): T = {
+    try {
+      val clazz = Class.forName(className)
+      println(s"Create instance of $clazz")
+      clazz.newInstance().asInstanceOf[T]
+    } catch {
+      case ex: Throwable =>
+        scala.concurrent.util.Unsafe.instance.allocateInstance(
+          Class.forName(className)
+        ).asInstanceOf[T]
+    }
+  }
+}
+
+class StaticJavaRuntime extends FunSuite {
+
+  implicit val staticOnly = static.StaticOnly
+
+  test("runtime instantiated pickler should work") {
+
+    val c = Container(List(1,2,3,4,5))
+    val containerPickler = implicitly[Pickler[Container[List[Int]]]]
+    val pickled = c.pickle.value
+
+    val picklerClassName = containerPickler.getClass.getName
+    val runtimePickler = RuntimeHelper.getInstance[Pickler[Container[Any]]](picklerClassName)
```

Comment:
That's actually what we're doing, although we don't use a ClassLoader. Is there any benefit from using it? It sounds to me like I have the same problem than before: I can't generalize this mechanism for both client and server (because the client won't execute the same operations as the server, they don't share any logic, and therefore I can't make them load the pickler classes when they boot up). In this concrete case, the client will serialize something and maybe load it into the classloader, but the server won't have anything in that classloader when unpickling because they're independent).
  
There's also another failed workaround that I tried before and I didn't tell you. My line of reasoning was the following: as I know that the reference to other implicit unpicklers (the captured unpickler, in this case) will be null when I create the unpickler reflectively, what I do is to instantiate first the captured unpickler using its classname and then set it as a field in the spore unpickler. The problem is that it doesn't work because at the same time the captured unpickler (let's say it's of type `Tuple2[Int, String]`) will depend on the unpicklers `Int` and `String`. When I tried it, I still got a NPE, and maybe in this case could be fixed because they are primitive types, but if the users decide to capture a more complicated type and give custom picklers/unpicklers, it won't work.